### DISCUSSION TITLE
Preventable event pattern helper function(s)

### DISCUSSION THREAD

**Discussion by 2conmeo** [rusianh/eShopProject_MVC_API#5] [Ideas]
The problem is always state. Whether or not to make components that are stateful or stateless is always a big debate. React's paradigm says always be stateless, but components are meant to encapsulate predictable functionality, so the more of that the component can provide, the less code needs be in consuming devs apps.

**Comment by rusianh** [comment]
pretty simple in implementation i think (there are probably things that could be added), but a pretty powerful method of having a component be somewhat stateful for simplicity but still allowing selective developer control to "go stateless" when necessary/beneficial.
